### ISSUE TITLE
AMD compatibility broken in 1.8.1.

### ISSUE THREAD

**Issue by mbostock** [mdaines/viz.js#130]
Viz.js 1.8.1 adds a new UMD block that 1.8.0 didn’t have:

```js
if (typeof exports === 'object' && typeof module === 'object')
  module.exports = Module;
else if (typeof define === 'function' && define['amd'])
  define([], function() { return Module; });
else if (typeof exports === 'object')
  exports["Module"] = Module;
```

But Viz.js 1.8.1 retains the global / CommonJS export block from previous releases:

```js
if (typeof module === "object" && module.exports) {
  module.exports = Viz;
} else {
  global.Viz = Viz;
}
```

As a result, in an AMD environment (such as when using [d3-require](https://github.com/d3/d3-require) inside of [Observable](https://beta.observablehq.com)), loading Viz.js 1.8.1 returns the `Module` object (created by Emscripten?) rather than the expected `Viz` object.

Replacing both blocks with something like this should work:

```js
if (typeof exports === 'object' && typeof module !== 'undefined') {
  module.exports = Viz;
} else if (typeof define === 'function' && define.amd) {
  define(function() { return Viz; });
} else {
  global.Viz = Viz;
}
```

(I often use Rollup to generate UMD bundles from ES modules; see [@observablehq/graphviz](https://github.com/observablehq/graphviz) for an example.)

**Comment by mbostock** [comment]
I see you’re already using Rollup for 2.0.0, hooray! So feel free to close this if you don’t feel a need to ship a patch release for v1. 😄

**Comment by mdaines** [comment]
@mbostock I've released 1.8.2, which I think should address this.

I am using Rollup for 2.0.0, but I'm also splitting the Emscripten stuff into a separate file so it can be used as a Web Worker. That doesn't quite work in Observable yet as I just found out. I've created an issue for that here: #131

**Comment by magjac** [comment]
Thanks a lot. Here is evidence that it's working:
https://beta.observablehq.com/@magjac/d3-graphviz-fiddling-not-working
https://talk.observablehq.com/t/global-is-undefined-when-using-viz-js-through-d3-graphviz/706/16?u=magjac
